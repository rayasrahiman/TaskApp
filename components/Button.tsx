import React, {CSSProperties} from 'react';
import {View, StyleSheet, Text, TouchableOpacity} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import {Colors} from '../constants/colors';

export type Props = {
  title: string;
  onPress(): void;
  btnWrapper: any;
  btnContainerProp: any;
  btnTxtProp: any;
};

const Button: React.FC<Props> = ({
  btnWrapper,
  title,
  onPress,
  btnContainerProp,
  btnTxtProp,
  ...touchable
}) => {
  return (
    <View style={{...styles.container, ...btnWrapper}}>
      <TouchableOpacity
        {...touchable}
        onPress={onPress}
        style={{...styles.buttonContainer, ...btnContainerProp}}>
        <Text style={{...styles.btnTxt, ...btnTxtProp}}>{title}</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginHorizontal: wp(5),
  },
  buttonContainer: {
    width: '100%',
    height: hp(7),
    borderRadius: 10,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.primary700,
  },
  btnTxt: {
    fontSize: hp(3),
    paddingVertical: hp(0.5),
    color: 'white',
  },
});

export default Button;
