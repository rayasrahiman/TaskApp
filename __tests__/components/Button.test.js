import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import {shallow} from 'enzyme';

import Button from '../../components/Button';

beforeEach(jest.spyOn(console, 'error').mockImplementation(() => undefined));
afterEach(() => jest.clearAllMocks());

describe('Button', () => {
  test('should render Button correctly', () => {
    const wrapper = render(<Button />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
