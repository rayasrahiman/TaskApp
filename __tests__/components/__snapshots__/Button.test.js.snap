// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button correctly 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 37.5,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0E6E96",
        "borderRadius": 10,
        "height": 93.5,
        "justifyContent": "center",
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 40,
          "paddingVertical": 6.5,
        }
      }
    />
  </View>
</View>
`;
